spynnaker8/models/connectors/multapse_connector.py

```diff
@@ -22,23 +23,29 @@ class MultapseConnector(_BaseClass):
     """
     __slots__ = []
 
-    def __init__(self, num_synapses, allow_self_connections=True,
-                 with_replacement=True, safe=True, verbose=False):
+    def __init__(self, n, allow_self_connections=True,
+                 with_replacement=True, safe=True, verbose=False,
+                 rng=None):
         super(MultapseConnector, self).__init__(
-            num_synapses=num_synapses,
-            allow_self_connections=allow_self_connections,
-            with_replacement=with_replacement, safe=safe, verbose=verbose)
+            num_synapses=n, allow_self_connections=allow_self_connections,
+            with_replacement=with_replacement, safe=safe, verbose=verbose,
+            rng=rng)
 
     def get_rng_next(self, num_synapses, prob_connect):
-        # This needs to be edited to work with PyNN 0.8+
-        # equivalent PyNN 0.7 call with the allowed "multinomial"
-        # returns an array, this with "binomial" only returns a single value...
+        # Below is how numpy does multinomial internally...
+        size = len(prob_connect)
+        multinomial = numpy.zeros(size, int)
+        total = 1.0
+        dn = num_synapses
+        for j in range(0, size - 1):
+            multinomial[j] = self._rng.next(
+                1, distribution="binomial",
+                parameters={'n': dn, 'p': prob_connect[j] / total})
+            dn = dn - multinomial[j]
+            if dn <= 0:
+                break
+            total = total - prob_connect[j]
+        if dn > 0:
+            multinomial[size - 1] = dn
 
```

Comment:
Ah, yes, I see - I wasn't that far off.  Knew there was probably a better way of doing it though!

Reply:
I basically copied this from numpy!